### PR TITLE
Support to PKCS12 keystores through encryption.keyStore.type property

### PR DESCRIPTION
**Pull request by bberto**
fixes gh-953

### CODE REVIEW

File: spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/encrypt/KeyProperties.java [hunk 1]
```diff
@@ -113,9 +113,9 @@ public void setKeyStore(KeyProperties.KeyStore keyStore) {
 		private String secret;
 
 		/**
-		 * The KeyStore type. Defaults to jks.
+		 * The KeyStore type (key store file extension used as default).
 		 */
-		private String type = "jks";
+		private String type;
```
Comment (ryanjbaxter) [inline spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/encrypt/KeyProperties.java:118]: Why did you remove the default value?

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]

File: spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/encrypt/KeyProperties.java (hunk 1 above)
Comment (bberto) [reply spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/encrypt/KeyProperties.java:118]: Currently this property is not used in `TextEncryptorUtils.createTextEncryptor` and the default behaviour is to infer the keystore type from the file extension.

Removing the default value this behaviour is preserved, avoiding a breaking change for who is using, for example, `encryption.key-store.location=classpath:test.jceks`  with no `encryption.key-store.type` set.

However, while writing this comment, I noticed a default value is expected here:
https://github.com/spring-cloud/spring-cloud-config/blob/e645c802157caa88a8ed50ecb9067335f0c7522f/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EncryptionAutoConfiguration.java#L104

What do you suggest?

Comment (ryanjbaxter) [reply spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/encrypt/KeyProperties.java:118]: Could you enable your added code below only if it is not the default value?

Comment (bberto) [reply spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/encrypt/KeyProperties.java]: I think it's better to rollback to default value. The current behaviour (infer type from extension) is not documented anywhere and the error in case of keystore type mismatch is pretty clear.

**Comment by bberto** [comment]
Sorry, I made a mess with this PR. I opened a new one:
https://github.com/spring-cloud/spring-cloud-commons/pull/963
